src/libstore/gc.cc

```diff
@@ -245,27 +246,27 @@ void LocalStore::findRoots(const Path & path, std::filesystem::file_type type, R
             else {
                 target = absPath(target, dirOf(path));
                 if (!pathExists(target)) {
-                    if (isInDir(path, stateDir + "/" + gcRootsDir + "/auto")) {
+                    if (isInDir(path, fs::path{stateDir.get()} / gcRootsDir / "auto")) {
                         printInfo("removing stale link from '%1%' to '%2%'", path, target);
                         unlink(path.c_str());
                     }
                 } else {
-                    if (!std::filesystem::is_symlink(target)) return;
+                    if (!fs::is_symlink(target)) return;
                     Path target2 = readLink(target);
                     if (isInStore(target2)) foundRoot(target, target2);
                 }
             }
         }
 
-        else if (type == std::filesystem::file_type::regular) {
+        else if (type == fs::file_type::regular) {
             auto storePath = maybeParseStorePath(storeDir + "/" + std::string(baseNameOf(path)));
             if (storePath && isValidPath(*storePath))
                 roots[std::move(*storePath)].emplace(path);
         }
 
     }
 
-    catch (std::filesystem::filesystem_error & e) {
+    catch (fs::filesystem_error & e) {
```

Comment:
We shouldn't throw a filesystem exception here either. Also needs be wrapped in a nix exception so we not loose the addTrace feature.